src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,173 @@
+import type { Options } from './index';
+
+const boolOpts = [
+  'makeCumulative',
+  'loop',
+  'showIcons',
+  'showGroups',
+  'mouseControls',
+  'keyboardControls',
+  'autorun',
+  'injectStyles',
+  'highlightBars',
+  'selectBars',
+  'fixedScale',
+] as const;
+const numberOpts = [
+  'labelsWidth',
+  'tickDuration',
+  'topN',
+  'minHeight',
+  'minWidth',
+  'marginTop',
+  'marginRight',
+  'marginBottom',
+  'marginLeft',
+] as const;
+const strOpts = ['theme', 'startDate', 'endDate'] as const;
+const strOrNumberOpts = ['colorSeed', 'inputHeight', 'inputWidth', 'height', 'width'] as const;
+const strOrFuncOpts = ['title', 'subTitle', 'dateCounter', 'caption'] as const;
+
+const validDataShapes = ['long', 'wide'];
```

Comment:
also more type-safety:
```suggestion
const validDataShapes: Array<Options['dataShape']> = ['long', 'wide'];
```